app/forms/providers/application_merits_task/statement_of_case_choice_form.rb

```diff
@@ -0,0 +1,47 @@
+module Providers
+  module ApplicationMeritsTask
+    class StatementOfCaseChoiceForm < BaseForm
+      form_for ::ApplicationMeritsTask::StatementOfCase
+
+      attr_accessor :statement, :upload, :typed, :choice
+
+      validate :validate_any_checkbox_checked, unless: :draft?
+      validates :statement, presence: true, if: :typed?
+
+      set_callback :save, :before, :update_values
+
+      def exclude_from_model
+        [:choice]
+      end
+
+      def initialize(params = {})
+        super
+        %i[typed upload].each do |choice|
+          value = params[choice].present? ? params[choice].include?("true") : params[:model].send(choice)
+          send("#{choice}=", value)
+        end
+      end
+
+    private
+
+      def update_values
+        attributes["upload"] = attributes["upload"].is_a?(Array) && attributes["upload"].include?("true")
+        attributes["typed"] = attributes["typed"].is_a?(Array) && attributes["typed"].include?("true")
+        attributes["statement"] = "" if attributes["typed"] == false
```

Comment:
does this need to check for `attributes["typed"] == ""` (for false/unchecked) as well?

Reply:
In test cases maybe but, in browser, this always gets cast to `true` or `false`, never stringified, missing or empty